geode-membership/src/integrationTest/java/org/apache/geode/distributed/internal/membership/gms/MembershipIntegrationTest.java

```diff
@@ -0,0 +1,206 @@
+package org.apache.geode.distributed.internal.membership.gms;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.concurrent.ExecutorService;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import org.apache.geode.distributed.internal.membership.api.MemberIdentifier;
+import org.apache.geode.distributed.internal.membership.api.MemberIdentifierFactoryImpl;
+import org.apache.geode.distributed.internal.membership.api.MemberStartupException;
+import org.apache.geode.distributed.internal.membership.api.Membership;
+import org.apache.geode.distributed.internal.membership.api.MembershipBuilder;
+import org.apache.geode.distributed.internal.membership.api.MembershipConfig;
+import org.apache.geode.distributed.internal.membership.api.MembershipConfigurationException;
+import org.apache.geode.distributed.internal.membership.api.MembershipLocator;
+import org.apache.geode.distributed.internal.membership.api.MembershipLocatorBuilder;
+import org.apache.geode.distributed.internal.tcpserver.TcpClient;
+import org.apache.geode.distributed.internal.tcpserver.TcpSocketCreator;
+import org.apache.geode.distributed.internal.tcpserver.TcpSocketCreatorImpl;
+import org.apache.geode.distributed.internal.tcpserver.TcpSocketFactory;
+import org.apache.geode.internal.serialization.DSFIDSerializer;
+import org.apache.geode.internal.serialization.internal.DSFIDSerializerImpl;
+import org.apache.geode.logging.internal.executors.LoggingExecutors;
+
+/**
+ * Tests of using the membership APIs to make multiple Membership systems that communicate
+ * with each other and form a group
+ */
+public class MembershipIntegrationTest {
+  private InetAddress localHost;
+  private DSFIDSerializer dsfidSerializer;
+  private TcpSocketCreator socketCreator;
+
+  @Rule
+  public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  @Before
+  public void before() throws IOException, MembershipConfigurationException {
+    localHost = InetAddress.getLocalHost();
+    dsfidSerializer = new DSFIDSerializerImpl();
+    socketCreator = new TcpSocketCreatorImpl();
+  }
+
+  @Test
+  public void oneMembershipCanStartWithALocator()
+      throws IOException, MemberStartupException {
+    final MembershipLocator<MemberIdentifier> locator = createLocator(0);
+    locator.start();
+
+    final Membership<MemberIdentifier> membership = createMembership(locator,
+        locator.getPort());
+    start(membership);
+
+    assertThat(membership.getView().getMembers()).hasSize(1);
+  }
+
+  @Test
+  public void twoMembersCanStartWithOneLocator()
```

Comment:
do we want to say `twoMembershhipsCanStart…` to be consistent w/ other test names?